src/SmartFormat/Utilities/ReflectionUtils.cs

```diff
@@ -0,0 +1,53 @@
+﻿//
+// Copyright SmartFormat Project maintainers and contributors.
+// Licensed under the MIT license.
+//
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SmartFormat.Utilities;
+internal static class ReflectionUtils
+{
+    #region *** Dictionary ***
+
+    public static readonly List<Type> DictionaryInterfaces = new() {
+        typeof(IDictionary<,>), // 1
+        typeof(IDictionary), // 2
+        typeof(IReadOnlyDictionary<,>) // 3
+    };
+
+    public static bool IsDictionary(Type type)
+    {
+        return DictionaryInterfaces
+            .Exists(dictInterface =>
```

Comment:
Can we do this without the Linq or maybe pass in a static `Func<>` to avoid the allocations?